userspace/libsinsp/filterchecks.cpp

```diff
@@ -4417,6 +4419,14 @@ uint8_t* sinsp_filter_check_event::extract(sinsp_evt *evt, OUT uint32_t* len, bo
 				{
 					m_u32val = 1;
 				}
+
+				if(m_field_id == TYPE_ISOPEN_EXEC && ( (flags & PPM_O_TMPFILE) || (flags & PPM_O_CREAT)) )
+				{
+					parinfo = evt->get_param(etype == PPME_SYSCALL_OPENAT_2_X ? 4 : 3);
+					ASSERT(parinfo->m_len == sizeof(uint32_t));
+					uint32_t mode_bits = *(uint32_t *)parinfo->m_val;
+					m_u32val = (mode_bits & (S_IRWXU | S_IXUSR | S_IRWXG | S_IXGRP | S_IRWXO | S_IXOTH))? 1 : 0;
```

Comment:
Looks like S_IXOTH is a linux-only thing. Wanna make the mask dependent on linux vs non linux?

Reply:
Checked and it seems to be available in OSx.